src/loaders/TextureLoader.js

```diff
@@ -36,6 +50,15 @@ Object.assign( TextureLoader.prototype, {
 			texture.format = isJPEG ? RGBFormat : RGBAFormat;
 			texture.needsUpdate = true;
 
+			texture.onUpdate = function () {
+
+				console.info( "Removing texture", texture.id, url );
+				Cache.remove( cacheKey );
```

Comment:
So here we remove the `imageBitmap` we stored in the cache because we are going to dispose all of its associated resources. Should we instead just never put it in the cache? Perhaps it doesn't matter, since you can call `texture.image.close()` and `delete texture.image` twice without throwing an error, and if we happen to make use of the cache in the window between load and `onUpdate` that seems good. Do we expect that ever to happen?

Reply:
Related question: Is it possible for two textures to share a url to an image and have the following issue:

Texture 1 loads image, puts {`url`, `ImageBitmap`} into the cache.
Texture 2 loads image, pulls from the cache.
Texture 1 is uploaded to the GPU, uses `texture.image.data`, calls `onUpdate` to `close` the image and remove it from the cache.
Texture 2 is uploaded to the GPU, tries to use `texture.image.data` but fails, because the image was closed.

Reply:
Do we ever expect to have to upload a texture to the GPU multiple times?

Reply:
Re: "Should we instead just never push it in the cache", yeah probably, did it this way mostly to keep the patch small and localized to avoid future merge issues, but like we discussed (and as you pointed out above) there could be an issue with textures sharing the same Image.

Reply:
Resolved this possible race condition by removing the image from the cache righ as onLoad is called. This happens immediately after it is added to the cache so there is no chance for the cached image to get picked up. Might be slightly cleaner to just not add the image to the cache, but again trying to keep the diff small here since I expect this change to move into Hubs once GLTFLoader hooks land.